src/Altinn.Apps/AppTemplates/AspNet/App/Startup.cs

```diff
@@ -85,6 +85,16 @@ public void ConfigureServices(IServiceCollection services)
             X509Certificate2 cert = new X509Certificate2(certPath);
             SecurityKey key = new X509SecurityKey(cert);
 
+            string hostName = string.Empty;
+            if (Environment.GetEnvironmentVariable("GeneralSettings__HostName") != null)
+            {
+                hostName = Environment.GetEnvironmentVariable("GeneralSettings__HostName");
+            }
+            else
+            {
+                hostName = Configuration["GeneralSettings:HostName"];
```

Comment:
You should only need this line because the configuration value with the double underscore should be overriden in Configuration. As long as services.AddEnvironmentVariables() or webhost is based on DefaultWebHost, all Environment variables with double underscore will also be put in Configuration and double underscores will be replaced by colon (:)